calico/felix/test/stub_fiptables.py

```diff
@@ -103,21 +111,36 @@ def __eq__(self, other):
 
         return True
 
+    def __str__(self):
+        output = self.target_name
+        if self.target_args:
+            output += " " + str(self.target_args)
+        output += " " + (self.protocol if self.protocol else "all")
+        output += " " + (self.src if self.src else "anywhere")
+        output += " " + (self.in_interface if self.in_interface else "any_in")
+        output += " " + (self.out_interface if self.out_interface else "any_out")
+        if self.match_name:
+            output += " " + self.match_name
+        output += ((" " + str(self.match_args)) if self.match_args else "")
+
+        return output
+
     def __ne__(self, other):
-        return not self.__eq__(self,other)
+        return not self.__eq__(other)
 
 class Chain(object):
     """
     Mimic of an IPTC chain. Rules must be a list (not a set).
     """
-    def __init__(self, name, rules=[]):
+    def __init__(self, name):
         self.name = name
-        self.rules = rules
+        self.rules = []
+        self.type = None # Not known until put in table.
 
     def flush(self):
-        self.rules = []
+        del self.rules[:]

```

Comment:
I agree that's possible, but it might be confusing because it is possible for a caller to get hold of rules from a chain (and in some cases the rules are directly accessed this way). It seems less confusing if `chain.rules` is a reference to the rules of the chain rather than a reference to the rules before they were flushed.